pandas/_libs/lib.pyx

```diff
@@ -650,22 +650,20 @@ ctypedef fused int6432_t:
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def array_equal_fast(
-    ndarray[int6432_t, ndim=1] left, ndarray[int6432_t, ndim=1] right,
-) -> bool:
+def indexer_equal_fast(ndarray[int6432_t, ndim=1] left, int n) -> bool:
```

Comment:
While renaming this function, maybe `is_range_indexer` might be more appropriate? Or at least clarifying that this only compares if the indexer is equal to `range(n)`

Reply:
Yeah thought about this myself and agree.

Reply:
Changed